geonode/contrib/monitoring/models.py

```diff
@@ -0,0 +1,1393 @@
+# -*- coding: utf-8 -*-
+#########################################################################
+#
+# Copyright (C) 2017 OSGeo
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+#########################################################################
+from __future__ import print_function
+
+import logging
+import types
+from urlparse import urlparse
+
+from socket import gethostbyname
+from datetime import datetime, timedelta
+from decimal import Decimal
+
+from django.db import models
+from django import forms
+from django.conf import settings
+from django.http import Http404
+from jsonfield import JSONField
+
+from django.utils.translation import ugettext_noop as _
+from django.core.urlresolvers import reverse
+from django.contrib.auth import get_user_model
+
+try:
+    from django.contrib.gis.geoip2 import GeoIP2 as GeoIP
+except ImportError:
+    from django.contrib.gis.geoip import GeoIP
+
+import user_agents
+from multi_email_field.forms import MultiEmailField
```

Comment:
needs to be added in requirements as it is rasing:

File "/home/vagrant/worldmap_28/geonode.git/geonode/contrib/monitoring/models.py", line 46, in <module>
    from multi_email_field.forms import MultiEmailField
ImportError: No module named multi_email_field.forms

I would suggest to git clone this branch from scratch and try running it in a clean virtualenv to make sure everything is in place

Reply:
Thx. good catch. I've added this to `setup.py` though. I'll add an issue to make `requirements.txt` as authoritative source of list of dependencies.

Reply:
oh thanks, I always rely on that when creating my virtualenv. Didn't notice it was in setup.py